tests/flow/test_multi_label.py

```diff
@@ -229,3 +229,33 @@ def test09_test_query_graph_populate_nodes_labels(self):
         self.env.assertEquals(query_result.nodes_created, 2)
         self.env.assertEquals(query_result.relationships_created, 2)
 
+    def test10_test_delete_label(self):
+        graph = Graph(self.redis_con, 'delete_multi_label')
+
+        query = """CREATE (a:L1) RETURN labels(a)"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.labels_added, 1)
+        self.env.assertEquals(query_result.nodes_created, 1)
+        self.env.assertEquals(query_result.result_set[0][0], ["L1"])
+
+        query = """CREATE (a:L2) RETURN labels(a)"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.labels_added, 1)
+        self.env.assertEquals(query_result.nodes_created, 1)
+        self.env.assertEquals(query_result.result_set[0][0], ["L2"])
+
+        query = """CREATE (a:L3) RETURN labels(a)"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.labels_added, 1)
+        self.env.assertEquals(query_result.nodes_created, 1)
+        self.env.assertEquals(query_result.result_set[0][0], ["L3"])
+
+        query = """MATCH (a) DELETE a"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.nodes_deleted, 3)
+
+        query = """CREATE (a:L4) RETURN labels(a)"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.labels_added, 1)
+        self.env.assertEquals(query_result.nodes_created, 1)
+        self.env.assertEquals(query_result.result_set[0][0], ["L4"])
```

Comment:
I'll assume without the fix this `a` node would have 2 labels ?

Reply:
yes

Reply:
Yes, and there may be more than one extra label.  

I have encountered a situation where a node is marked with three or even four labels